conans/client/graph/graph_builder.py

```diff
@@ -54,20 +53,24 @@ def extend_build_requires(self, graph, node, build_requires_refs, check_updates,
 
         conanfile = node.conanfile
         scope = conanfile.display_name
-        requires = [Requirement(ref) for ref in build_requires_refs]
+        requires = []
+        for ref, context in build_requires_refs:
+            r = Requirement(ref)
```

Comment:
may be add ctor which has both ref and context?

Reply:
Please, do not enter into details yet, there will be plenty of time to review the PR 😉  Better start with the look-and-feel of the tests (and take into account that this is under heavy development and I might change the implementation at any moment)